pkg/builder/csolution/builder.go

```diff
@@ -603,6 +603,8 @@ func (b CSolutionBuilder) build() (err error) {
 		return err
 	}
 
```

Comment:
When `--no-database` is used, probably there is no need to call `b.needRebuild()` ?
```suggestion
if !b.Options.NoDatabase {
	b.Options.Rebuild, err = b.needRebuild()
	if err != nil {
		log.Error(err)
		return err
	}
}
```

Reply:
In this case, `b.Options.Rebuild` will retain its previous state. If it is set to `true`, the rebuild process will still occur regardless. 

We can set it before and update it later 
```
	b.Options.Rebuild = !b.Options.NoDatabase

	if b.Options.Rebuild {
		rebuildNeeded, err := b.needRebuild()
		if err != nil {
			log.Error(err)
			return err
		}
		b.Options.Rebuild = rebuildNeeded
	}
```